seacrowd/sea_datasets/onto4all/onto4all.py

```diff
@@ -0,0 +1,162 @@
+"""
+
+_HOMEPAGE = "https://huggingface.co/datasets/ontocord/onto4all"
+
+_LANGUAGES = ["vie"] 
+
+_LICENSE = Licenses.CC0_1_0.value
+
+_LOCAL = False
+
+_URLS = "https://huggingface.co/datasets/ontocord/onto4all/resolve/main/data/train-00000-of-00001.parquet?download=true"
+
+_SUPPORTED_TASKS = [Tasks.QUESTION_ANSWERING]
+
+_SOURCE_VERSION = "1.0.0"
+
+_SEACROWD_VERSION = "1.0.0"
+
+class Onto4AllDataset(datasets.GeneratorBasedBuilder):
+    """Onto4All is a subsample of other open source performant conversational datasets. We start with a carefully curated subset of the OpenHermes-2.5-Viet dataset, co-created by @qnguyen3 and @Teknium. This dataset is specifically designed to support the training and evaluation of Multilingual language models, such as Vistral-7B-chat and VinaLlama-7B-chat, and is derived from our Supervised Fine-Tuning (SFT) data. We have included Vietnamese here, but will add more languages."""
+
+    SOURCE_VERSION = datasets.Version(_SOURCE_VERSION)
+    SEACROWD_VERSION = datasets.Version(_SEACROWD_VERSION)
+
+    BUILDER_CONFIGS = [
+        SEACrowdConfig(
+            name=f"{_DATASETNAME}_source",
+            version=SOURCE_VERSION,
+            description=f"{_DATASETNAME} source schema",
+            schema="source",
+            subset_id=f"{_DATASETNAME}",
+        ),
+        SEACrowdConfig(
+            name=f"{_DATASETNAME}_seacrowd_qa",
+            version=SEACROWD_VERSION,
+            description=f"{_DATASETNAME} SEACrowd schema",
+            schema="seacrowd_qa",
+            subset_id=f"{_DATASETNAME}",
+        ),
+    ]
+
+    DEFAULT_CONFIG_NAME = f"{_DATASETNAME}_source"
+
+    def _info(self) -> datasets.DatasetInfo:
+
+        if self.config.schema == "source":
+            features = datasets.Features(
+                {
+                    "id": datasets.Value("int32"),
+                    "type": datasets.Value("string"),
+                    "conversation": datasets.Sequence({
+                        "from": datasets.Value("string"),
+                        "value": datasets.Value("string"),
+                        "weight": datasets.Value("int32"),
+                    })
+                }
+            )
+
+        elif self.config.schema == "seacrowd_qa":
+            features = schemas.qa_features
+
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=features,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager: datasets.DownloadManager) -> List[datasets.SplitGenerator]:
```

Comment:
Thanks @patrickamadeus for the submission.

I couldn't run the unit tests `python -m tests.test_seacrowd seacrowd/sea_datasets/onto4all/onto4all.py --subset_id onto4all`. Here's the error I receive.

```
======================================================================
ERROR: runTest (__main__.TestDataLoader)
Run all tests that check:
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/yong/.pyenv/versions/3.8.10/lib/python3.8/site-packages/datasets/builder.py", line 1687, in _prepare_split_single
    example = self.info.features.encode_example(record) if self.info.features is not None else record
  File "/Users/yong/.pyenv/versions/3.8.10/lib/python3.8/site-packages/datasets/features/features.py", line 1866, in encode_example
    return encode_nested_example(self, example)
  File "/Users/yong/.pyenv/versions/3.8.10/lib/python3.8/site-packages/datasets/features/features.py", line 1243, in encode_nested_example
    {
  File "/Users/yong/.pyenv/versions/3.8.10/lib/python3.8/site-packages/datasets/features/features.py", line 1244, in <dictcomp>
    k: encode_nested_example(sub_schema, sub_obj, level=level + 1)
  File "/Users/yong/.pyenv/versions/3.8.10/lib/python3.8/site-packages/datasets/features/features.py", line 1277, in encode_nested_example
    for k, (sub_schema, sub_objs) in zip_dict(schema.feature, obj):
  File "/Users/yong/.pyenv/versions/3.8.10/lib/python3.8/site-packages/datasets/utils/py_utils.py", line 323, in zip_dict
    yield key, tuple(d[key] for d in dicts)
  File "/Users/yong/.pyenv/versions/3.8.10/lib/python3.8/site-packages/datasets/utils/py_utils.py", line 323, in <genexpr>
    yield key, tuple(d[key] for d in dicts)
IndexError: only integers, slices (`:`), ellipsis (`...`), numpy.newaxis (`None`) and integer or boolean arrays are valid indices

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/yong/Desktop/seacrowd-datahub/tests/test_seacrowd.py", line 134, in setUp
    self.dataset_source = datasets.load_dataset(
  File "/Users/yong/.pyenv/versions/3.8.10/lib/python3.8/site-packages/datasets/load.py", line 2152, in load_dataset
    builder_instance.download_and_prepare(
  File "/Users/yong/.pyenv/versions/3.8.10/lib/python3.8/site-packages/datasets/builder.py", line 948, in download_and_prepare
    self._download_and_prepare(
  File "/Users/yong/.pyenv/versions/3.8.10/lib/python3.8/site-packages/datasets/builder.py", line 1711, in _download_and_prepare
    super()._download_and_prepare(
  File "/Users/yong/.pyenv/versions/3.8.10/lib/python3.8/site-packages/datasets/builder.py", line 1043, in _download_and_prepare
    self._prepare_split(split_generator, **prepare_split_kwargs)
  File "/Users/yong/.pyenv/versions/3.8.10/lib/python3.8/site-packages/datasets/builder.py", line 1549, in _prepare_split
    for job_id, done, content in self._prepare_split_single(
  File "/Users/yong/.pyenv/versions/3.8.10/lib/python3.8/site-packages/datasets/builder.py", line 1706, in _prepare_split_single
    raise DatasetGenerationError("An error occurred while generating the dataset") from e
datasets.builder.DatasetGenerationError: An error occurred while generating the dataset

----------------------------------------------------------------------
Ran 1 test in 0.509s

FAILED (errors=1)
```

I think the bug comes from this function. Could you cehck if data is properly extracted?

Reply:
Hi @TysonYu , it is working fine from my side. Could you please pull from the latest commit and see if it's working good right now?

Thank you 😊

Reply:
That works now. Thank you